wasmtime-environ/src/cranelift.rs

```diff
@@ -224,8 +224,10 @@ impl crate::compilation::Compiler for Cranelift {
                         }
 
                         let mut trans = FuncTranslator::new();
+                        let wasm_types = WasmTypesMap::new();
```

Comment:
This creates a new empty `WasmTypesMap` for each function; to support multi-value, I think we need to get the `WasmTypesMap` that was populated when decoding the wasm type section.

Reply:
Ok, let me hunt around for that (wasn't paying too close attention to the purpose of the API change).

Reply:
See this commit: https://github.com/CraneStation/wasmtime/pull/399/commits/82e3b95dca82a87f33c5e412475bc814aefb9f8c

`translate_module` returns a `ModuleTranslationState` (ne `WasmTypesmap`) that contains the module's type signatures which are required when translating multi-value blocks.

Reply:
Aha! That's what I'm missing. Thanks 😄.